src/app/core/ui/menu-item-list-editor/menu-item-list-editor.component.ts

```diff
@@ -0,0 +1,212 @@
+import { Component, EventEmitter, Input, Output, inject } from "@angular/core";
+import { MatButton } from "@angular/material/button";
+import {
+  CdkDragDrop,
+  DragDropModule,
+  moveItemInArray,
+  transferArrayItem,
+} from "@angular/cdk/drag-drop";
+import { FaIconComponent } from "@fortawesome/angular-fontawesome";
+import { v4 as uuid } from "uuid";
+import { MatDialog } from "@angular/material/dialog";
+import { Logging } from "../../logging/logging.service";
+import { MenuItem } from "../navigation/menu-item";
+import { AdminMenuItemComponent } from "../../admin/admin-menu/admin-menu-item/admin-menu-item.component";
+import {
+  MenuItemForAdminUi,
+  MenuItemForAdminUiNew,
+} from "../../admin/admin-menu/menu-item-for-admin-ui";
+import { AdminMenuItemDetailsComponent } from "../../admin/admin-menu/admin-menu-item-details/admin-menu-item-details.component";
+
+/**
+ * A reusable component for editing lists of menu items with drag & drop,
+ * add/remove functionality. Used by both AdminMenuComponent and
+ * ShortcutDashboardSettingsComponent.
+ */
+@Component({
+  selector: "app-menu-item-list-editor",
+  standalone: true,
+  imports: [AdminMenuItemComponent, MatButton, DragDropModule, FaIconComponent],
+  templateUrl: "./menu-item-list-editor.component.html",
+  styleUrl: "./menu-item-list-editor.component.scss",
+})
+export class MenuItemListEditorComponent {
+  private readonly dialog = inject(MatDialog);
+
+  @Input() items: MenuItemForAdminUi[] = [];
+  @Input() showAddButton: boolean = true;
+  @Input() addButtonLabel: string = "Add New";
+  @Input() containerId: string = "menu-item-list-container";
+  @Input() itemType: string = "Menu Item";
+  @Input() allowSubMenu: boolean = true;
+  @Input() excludeNavigationItems: boolean = false;
```

Comment:
can you add JSDocs here as well, for those properties that are not completely self-explanatory (`showAddButton` and the like are probably already clear without comment)